plutus-wallet-api/ledger/Ledger/Typed/Scripts.hs

```diff
@@ -33,20 +45,49 @@ type ValidatorType (a :: Type) = DataType a -> RedeemerType a -> Validation.Pend
 type WrappedValidatorType = Data -> Data -> Data -> ()
 
 -- | The type of a connection.
-data ScriptInstance (a :: Type) where
+data ScriptConnection (a :: Type) where
     Validator
         :: ScriptType a
         => CompiledCode (ValidatorType a)
         -> CompiledCode (ValidatorType a -> WrappedValidatorType)
-        -> ScriptInstance a
+        -> ScriptConnection a
```

Comment:
I was thinking to just add more fields to this, and then have a smart constructor. This works too, I'm just not wild about `ScriptConnection` as a name...

Reply:
I deleted the `ScriptConnection` type. We don't actually use the `CompiledCode` values for anything other than getting the `ValidatorScript` and the `Address`.

Reply:
Yeah, it's really only an intermediate stage.